### DISCUSSION TITLE
Report | Week 03

### DISCUSSION THREAD

**Discussion by sfrancisw** [SE-TINF22B6/time2shine#22] [Show and tell]
# Updates
Hello everyone!
This week we worked on the following topics:

## Change of plans
We thought a bit about the scope of our project and made some changes. Instead of the durak game, we now want to implement black jack as our first game. The simple reason for that is that black jack has fewer and simpler rules and doesn't need as much visual work as durak does. This makes it a perfect choice for our first game!

## Tool Upgrades
We also found a promising tool that will help us create the games and visuals for our webapp. The tool we want to try out is called PixiJS, which is a 2D WebGL renderer. You can learn more about it on their [website](https://pixijs.com/).

## UML Diagramm
We created our first prototype of our [UML Diagramm](https://github.com/SE-TINF22B6/time2shine/blob/main/uml.drawio).
![image](https://github.com/SE-TINF22B6/time2shine/assets/122756244/1d99b73e-5c79-4d2f-9660-1e3836c778d2)

## UI Mockup
We created some simple mockups for our main website screens. This includes the account, game selection, leaderboard and the black jack screen for our first game.
![grafik](https://github.com/SE-TINF22B6/time2shine/assets/49996344/3ba78f23-b804-45c0-98b4-a9ef78d27787)

## SRS File
We also worked on our srs.md file, which you can find [here](https://github.com/SE-TINF22B6/time2shine/blob/main/srs.md).

---
That's it for this week! Stay tuned for further updates.
Your **favorite** team time2shine!

**Comment by LPkkjHD** [comment]
Thank you for the update! I'm thrilled to see the blackjack game being implemented first (always used to be a sucker for this game especially).
Also the detailed overview with the UML diagram makes a lot of sense and helps with understanding the structure.
I'd love to see more of the fancy webgl library you've chosen.

**Comment by fehlend0** [comment]
Hi team time2shine,

switching to blackjack as your first game sounds like a reasonable decision. It's true that blackjack has simpler rules and may require less visual work compared to Durak. It's always a good idea to start with something manageable and build from there. As for your tool upgrade, exploring PixiJS for game and visual creation is a positive step. It's essential to find the right tools that suit your project's needs, and PixiJS seems to be very promising. We also liked the prototype UML diagram. It looks good from the image you provided. And it's great to see that you're working on your SRS-file and your mockup. The SRS documentation is crucial for clear project communication and understanding and the mockups will help you visualize the project better and make any necessary adjustments early on.

It's great to see the progress you've made!

Your Statify Team
